sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/timeExpressions.scala

```diff
@@ -952,19 +894,18 @@ case class TimeFromMicros(child: Expression)
   group = "datetime_funcs")
 // scalastyle:on line.size.limit
 case class TimeToSeconds(child: Expression)
-  extends UnaryExpression with ImplicitCastInputTypes with CodegenFallback {
+  extends UnaryExpression with RuntimeReplaceable with ImplicitCastInputTypes {
+
+  override def replacement: Expression = StaticInvoke(
+    classOf[DateTimeUtils.type],
+    DecimalType(14, 6),
```

Comment:
```suggestion
    dataType,
```
to avoid duplication